zetaclient/mode/chaos/chaos.go

```diff
@@ -0,0 +1,78 @@
+// Package chaos provides chaos-wrappers for the TSS signer, for the zetacore client, and for
+// the standard clients of the connected chains.
+//
+// A chaos-wrapper overrides methods from the underlying client that return at least one error,
+// that is, they override methods that can fail. The wrapper methods may call their inner
+// counterparts or, depending on configured failure percentages, return ErrChaos.
+package chaos
+
+//go:generate go run generate/main.go
+//go:generate gofmt -w generated.go
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/rand"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog"
+
+	"github.com/zeta-chain/node/zetaclient/config"
+)
+
+var (
+	// ErrChaos is the error that gets returned by the wrapped methods when they fail.
+	ErrChaos = errors.New("chaos error")
+
+	ErrNotChaosMode     = errors.New("not in chaos mode")
+	ErrReadPercentages  = errors.New("failed to read chaos percentages")
+	ErrParsePercentages = errors.New("failed to parse chaos percentages")
+)
+
+// Source is the base chaos object from which all chaos interface implementations inherit.
+// It is safe for concurrent use by multiple goroutines.
+type Source struct {
```

Comment:
It would be good to have a more detailed comment on how the percentages and rand are being used , either here or above the shouldFail method